src/openms/source/CHEMISTRY/ElementDB.cpp

```diff
@@ -560,10 +562,9 @@ namespace OpenMS
   void ElementDB::clear_()
   {
     // names_ has the union of all Element*, deleting this is sufficient to avoid mem leaks
-    map<string, const Element*>::iterator it = names_.begin();
-    for (; it != names_.end(); ++it)
+    for (auto &[string,Element] : names_)
```

Comment:
better not to use a class name as variable name
    for (auto &[s, element] : names_)

But I just realized that this might lead to an unused variable warning.
Maybe better to just use
for (auto it = names_.begin(); it != names_.end(); ++it)